### ISSUE TITLE
A strange phenomenon

### ISSUE THREAD

**Issue by Billy-ZTB** [xwjabc/hed#60]
Hello! Thanks for providing the code, I discovered a strange thing while I was using this repo to do boundary detection. It was something about the label images. My label images was generated from semantic segmentation labels by discovering boundaries of different objects. Things went well when I use boundary labels generated by canny operation in cv2,  but when I used seg2edge function in matlab to generate boundary with radius=2（to get thicker boundary), the loss didn't decrease and the model couldn't learn boundary information. The loss function I was using is weighted_cross_entropy_loss in this repo. I have no idea how the boundary's thickness would influence the training process, could you please provide your opinion? 

Using edge from canny:
![canny](https://github.com/xwjabc/hed/assets/52749583/20cb7726-f9e8-4e6f-ae24-fdf81542cb29)
![image](https://github.com/xwjabc/hed/assets/52749583/88f4032f-c137-4262-9f43-9679f0712d63)
It can be seen that the loss is decreasing gradually.

Using thicker boundary:
![radius=2](https://github.com/xwjabc/hed/assets/52749583/518797e5-54d8-4389-95cd-e5e397faf934)
![image](https://github.com/xwjabc/hed/assets/52749583/c68fd9be-4ff7-4b40-a507-4fb1185ca87d)
![batch-109-1st-image](https://github.com/xwjabc/hed/assets/52749583/312660de-3020-45dc-a336-5f606c65212a)
Loss wouldn't decrease, it goes 4 times bigger than that in epoch1 after several epochs. Both two types of boundaries are binary boundary(edge pixels=1, non-edge=0). I was using pretrained parameters 'hed_pretrained_bsds.py36pickle'.

I am new in this field and I am not familiar with the mechanism of binary cross entropy loss. Could you please give me some hint? Looking forward to your reply! 😊
